src/ng2-datepicker/ng2-datepicker.component.ts

```diff
@@ -74,9 +75,9 @@ export class DatePickerOptions {
     this.initialDate = obj && obj.initialDate ? obj.initialDate : null;
     this.firstWeekdaySunday = obj && obj.firstWeekdaySunday ? obj.firstWeekdaySunday : false;
     this.format = obj && obj.format ? obj.format : 'YYYY-MM-DD';
-    this.selectYearText = obj && obj.selectYearText ? obj.selectYearText : 'Select Year';
-    this.todayText = obj && obj.todayText ? obj.todayText : 'Today';
-    this.clearText = obj && obj.clearText ? obj.clearText : 'Clear';
+    this.selectYearText = obj && obj.selectYearText ? obj.selectYearText : (locale)? locale.selectYearText() : 'Select Year';
+    this.todayText = obj && obj.todayText ? obj.todayText : (locale)? locale.todayText() : 'Today';
+    this.clearText = obj && obj.clearText ? obj.clearText : (locale)? locale.clearText() : 'Clear';
```

Comment:
maybe it would better to open the ILocale interface to let users define their own locales and create the DatePickerOptions object with their custom locales.